cardano-db/src/Cardano/Db/Migration/Extra/CosnumedTxOut/Queries.hs

```diff
@@ -202,45 +224,139 @@ countConsumed = do
 
-deleteConsumedTxOut :: forall m. MonadIO m => Trace IO Text -> Word64 -> ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut ::
+  forall m.
+  (MonadIO m, MonadBaseControl IO m) =>
+  Trace IO Text ->
+  Word64 ->
+  ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut trce blockNoDiff = do
+  maxTxInId <- findMaxTxInId blockNoDiff
+  case maxTxInId of
+    Left _ -> do
+      createConsumedTxOut
+      liftIO $ logInfo trce "Created ConsumedTxOut as there were no blocks present"
+    Right mxtxid -> do
+      migrateNextPage mxtxid False 0
+  where
+    migrateNextPage :: TxInId -> Bool -> Word64 -> ReaderT SqlBackend m ()
+    migrateNextPage mxTxInId ranCreateConsumedTxOut offst = do
+      pageEntries <- getInputPage offst pageSize
+      resPageEntries <- splitAndProcessPageEntries trce ranCreateConsumedTxOut mxTxInId pageEntries
+      when (fromIntegral (length pageEntries) == pageSize) $
+        migrateNextPage mxTxInId resPageEntries $!
+          offst
+            + pageSize
+
+-- Split the page entries by maxTxInId and proces
+splitAndProcessPageEntries ::
+  forall m.
+  (MonadIO m, MonadBaseControl IO m) =>
+  Trace IO Text ->
+  Bool ->
+  TxInId ->
+  [(TxInId, TxId, Word64)] ->
+  ReaderT SqlBackend m Bool
+splitAndProcessPageEntries trce ranCreateConsumedTxOut maxTxInId pageEntries = do
+  let entriesSplit = span (\(txInId, _, _) -> txInId <= maxTxInId) pageEntries
+  case entriesSplit of
+    -- empty lists just return
+    ([], []) -> pure True
+    -- the whole list is less that maxTxInId
+    (xs, []) -> do
+      deleteEntries xs
+      pure False
+    -- the whole list is greater that maxTxInId
+    ([], ys) -> do
+      shouldCreateConsumedTxOut ranCreateConsumedTxOut
+      updateEntries ys
+      pure True
+    -- the list has both bellow and above maxTxInId
+    (xs, ys) -> do
+      deleteEntries xs
+      shouldCreateConsumedTxOut ranCreateConsumedTxOut
+      updateEntries ys
+      pure True
+  where
+    deleteEntries = deletePageEntries
+    -- this is deleting one entry at a time to check in benchmarking
+    -- deleteEntries = mapM_ (\(_, txId, index) -> deleteTxOutConsumed txId index)
+    updateEntries = mapM_ (\(txInId, txId, index) -> updateTxOutConsumedByTxInIdUnique txId index txInId)
+
+    shouldCreateConsumedTxOut rcc =
+      unless rcc $ do
+        liftIO $ logInfo trce "Created ConsumedTxOut when handling page entries."
+        createConsumedTxOut
+
+-- this builds up a single delete query using the pageEntries list
+deletePageEntries ::
+  MonadIO m =>
+  [(TxInId, TxId, Word64)] ->
+  ReaderT SqlBackend m ()
+deletePageEntries transactionEntries = do
+  delete $ do
+    txOut <- from $ table @TxOut
+    where_
+      ( foldl1
+          (||.)
+          ( map
+              ( \(_, txId, index) ->
+                  txOut E.^. TxOutTxId E.==. val txId E.&&. txOut E.^. TxOutIndex E.==. val index
+              )
+              transactionEntries
+          )
+      )
```

Comment:
@kderme seems to work but would need double checking and benchmarking with original I'd say!

Reply:
I tried this on mainnet, but got an out of memory sql error. Same when I used a 10_000 page size.
I also tried this approach 
```
  deleteWhere [FilterOr $ mkFilter <$> transactionEntries]
  where
    mkFilter (_, txOutId, index) = FilterAnd [TxOutTxId ==. txOutId, TxOutIndex ==. index]
```
but eventually I got faster results by simply deleting the entries one by one.